components/feature/addons/src/main/java/mozilla/components/feature/addons/update/db/UpdateAttemptDao.kt

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.addons.update.db
+
+import androidx.room.Insert
+import androidx.room.Query
+import androidx.room.Dao
+import androidx.room.OnConflictStrategy
+
+/**
+ * Internal dao for accessing and modifying add-on update requests in the database.
+ */
+@Dao
+internal interface UpdateAttemptDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertOrUpdate(entity: UpdateAttemptEntity): Long
+
+    @Query("SELECT * FROM update_attempts where addon_id =:addonId")
+    fun getUpdateAttemptBy(addonId: String): UpdateAttemptEntity?
```

Comment:
nit: `getUpdateAttemptFor`. This could also be `getLastUpdateAttemptFor` since we're only storing the last attempt. That's why I thought about calling this "result", but this is OK too.

Reply:
👍